src/actions/receive-profile.js

```diff
@@ -94,6 +95,16 @@ export function viewProfile(
       selectedThreadIndex,
       pathInZipFile,
     });
+
+    // Update ProfileSharingStatus from profile metadata.
+    // If the profiler is not published before, networkURLsRemoved
+    // will be undefined.
```

Comment:
Iterating all the markers would be done only once at load time. I agree this can be some work but also this is done once.

But if we think about it moe broadly, there are profiles with no network markers, but there aren't profiles without any URL (think the script URLs). So we can say that we display the button always, but in the (short) future we'll want this button to remove all the URLs, not only the URLs in the network markers.

In that case let's not do the extra work but instead plan removing all the URLs when "sharing without URLs".

What do you think?

Reply:
Yeah, it sound good to me. Filed issue #1125 for this.